### PR TITLE
fix(config): Rename global `expire_metrics` to `expire_metrics_secs`

### PR DESCRIPTION
**Pull request by bruceg**
All configuration that represents a number of seconds should be named `_secs`,
but the global `expire_metrics` setting did not reflect that pattern. This
renames the setting and adds a deprecation warning for the old setting, which
will be removed in a later version.

Ref: https://github.com/vectordotdev/vector/pull/14224#pullrequestreview-1092276407

### CODE REVIEW

File: src/topology/mod.rs [hunk 1]
```diff
@@ -79,11 +79,28 @@ pub async fn start_validated(
 ) -> Option<(RunningTopology, mpsc::UnboundedReceiver<()>)> {
     let (abort_tx, abort_rx) = mpsc::unbounded_channel();
 
+    let expire_metrics = match (
+        config.global.expire_metrics,
+        config.global.expire_metrics_secs,
+    ) {
+        (e @ Some(_), None) => {
+            warn!(
+                "Global `expire_metrics` setting is deprecated and will be removed in a future version. Use `expire_metrics_secs` instead."
```
Comment (jszwedko) [inline src/topology/mod.rs:88]: Could we add this deprecation to the 0.25.0 upgrade guide too?

Big 👍 on Vector logging this though.

In https://github.com/vectordotdev/vector/blob/master/docs/DEPRECATION.md#deprecation we say that any deprecation messages logged by Vector should lead with `DEPRECATED` to make them easy to spot / grep for.

File: website/cue/reference/configuration.cue [hunk 2]
```diff
@@ -37,7 +37,7 @@ configuration: {
 			}
 		}
 
-		expire_metrics: {
+		expire_metrics_secs: {
```
Comment (jszwedko) [inline website/cue/reference/configuration.cue:40]: Technically according to https://github.com/vectordotdev/vector/blob/master/docs/DEPRECATION.md#deprecation we should leave the old option too and document it as deprecated.

**Review by jszwedko — COMMENTED** [review]
Thanks for this! Just a couple of notes below.

File: website/cue/reference/configuration.cue (hunk 2 above)
Comment (spencergilbert) [reply website/cue/reference/configuration.cue:40]: I'd personally be ok with not including it in the docs given the situation, but yeah - letter of the spec we should.

**Review by spencergilbert — COMMENTED** [review]
Agree with Jesse's review

**Review by spencergilbert — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Thanks for the follow-ups @bruceg !

**Pull request merged**